composer/core/data_spec.py

```diff
@@ -246,6 +248,9 @@ def _default_get_num_samples_in_batch(self, batch: Batch) -> int:
         if isinstance(batch, torch.Tensor):
             return batch.shape[0]
 
+        if isinstance(batch, BatchEncoding):
+            batch = dict(batch)
```

Comment:
can we do if transformer is installed, check here? You can do somethin glike:

```
try:
  from transformers.tokenization_utils_base import BatchEncoding
  _TRANSFORMERS_INSTALLED = True
except:
  _TRANSFORMERS_INSTALLED = False
```
at top and then gate check on that if statement

Reply:
done